### ISSUE TITLE
How to send message from server to client by websocket immediately?

### ISSUE THREAD

**Issue by Waching** [embedthis/appweb#499]
# Environment:
Appweb version: 4.7.1
O/S: Linux 
CPU: arm
Protocol: websocket
# Problem description:
I have used **httpSend** to send message from server to client at short intervals.

But I found that sometimes client received several messages in one package and sometimes it received a part of message in a package and the other part of the message in next package.

I think the messages must be buffered by server.
So I want to send the message to client immediately without buffered by server.
Then I called **httpFinalize**, the result is the client just received the first message from server and the left messages cannot be received.

I have no idea about this, could you please tell me which function can I call to meet my goal?

**Comment by embedthis** [comment]
The data provided to httpSend is buffered because web socket packets need to be created. httpSend will start to send the data immediately, unless the socket is already full. It should automatically send all the data. You should not need to do anything to get the data to fully drain.

Can you create a simple test case that demonstrates this. We've never seen it.

**Comment by Waching** [comment]
I've resolved this issue. I found that httpSend called httpSendBlock with HTTP_BUFFER which can cause this issue. I cloned a function from httpSend to call httpSendBlock with HTTP_BLOCK and this new function can resolve this issue.

The test case is simple, we send a short message from server to client every time but very quickly and continuously. Maybe the each time's sending interval is too short, the websocket considered them as one message then caused this issue.
